### DISCUSSION TITLE
When i use copy command there is no linebreak in copied text

### DISCUSSION THREAD

**Discussion by Archie-2021** [ahrm/sioyek#853] [Q&A] [answered]
When I copy a selection via `<C-c>` there is no linebreak in it. I use something like Zathura and i works fine on the same file. Am I doing something wrong here?
I am on a arch based (EOS) Linux distro.

Thanks

**Comment by ahrm** [comment]
This is by design, I think this is better than the alternative (though they may be some situations where the alternative is preferable).

**Reply by Archie-2021** [reply]
I didn't expect that... Can you please add such an option. I like to copy past from PDF files in my notes beside referencing them and this ways is very non-productive for me to do that. I think it makes sense to at least give it as an option. Altho I think it is very easy to remove linebreaks in clipboard via an script (I have one) but it is not possible to know where the original line where and join them automatically like that.

Thanks

**Reply by KrappRamiro** [reply, marked as answer]
For anyone coming here from internet, this was added in a future version, see:

https://github.com/ahrm/sioyek/issues/1189#issuecomment-2408864786

All you need to do is set the option `add_newlines_when_copying_text` to 1, see https://github.com/ahrm/sioyek?tab=readme-ov-file#configuration on how to do it

If you are using home-manager, all you need to do is set `"programs.sioyek.configadd_newlines_when_copying_text" = "1"`

Of course, this only applies to the development version, so you also need to use the development branch and build it, like this example

```nix
{
  lib,
  config,
  pkgs,
  ...
}: {
  options = {
    sioyek.enable = lib.mkEnableOption "Enables sioyek, a pdf reader";
  };

  config = lib.mkIf config.sioyek.enable {
    programs.sioyek = {
      enable = true;

      # We want the development branch of sioyek, it has the newest features
      package = pkgs.sioyek.overrideAttrs (oldAttrs: {
        pname = "sioyek-dev";
        version = "dev-2026-02-01";

        src = pkgs.fetchFromGitHub {
          owner = "ahrm";
          repo = "sioyek";
          rev = "development";
          hash = "sha256-hlMTBlJK6bkiRJS+NtXtZMolX3R0S/XE5dg2uwwDLBQ=";
        };
      });

      bindings = {
        "screen_down" = ["<C-d>"];
        "screen_up" = ["<C-u>"];
      };

      config = {
        "add_newlines_when_copying_text" = "1";
      };
    };
  };
}
```

**Reply by KrappRamiro** [reply]
@ahrm can this be marked as answered?
